README.md

```diff
@@ -49,34 +48,22 @@ Once the mutation succeeds, we could request apollo to send a refetch request to
 Optimistic update allows us to write predictable data into cache if we know ahead what it will be. This allows us to provide quicker experience feedback for our users. Eg. in an address book when the user marks an address as favourite, we know the mutation will respond with true but the mutation might take a few hundred ms to respond. So we use optimistic response to mark the address as favourite and show quicker UI feedback to our users. When the optimistic response and the actual mutation response match, apollo will ignore the response. If the actual response is different from optimistic response apollo will go through usual failure flow.
 
 - use optimistic response only for edits where you can predict the response well
-- don't use optimistic response for creates since you will need an id to insert into cache that you only usually get in mutation response
+- don't use optimistic response for creates since you will need an id to insert into cache that you only usually get in mutation response. You can work around this with extra work, by either generating ids in the FE (if possible) - or by optimistically adding new entities with temporary ids, and then removing those entities once the mutation returns.
 
 ### Cache update
 
 Cache updates are how we insert data into apollo cache after successful mutation response.
 
-There are usually two ways to go about this —
+When a mutation completes, we want to return all fields changed by the mutation. Usually this is as simple as returning the changed properties on the changed entity on updates, or the newly created entity on creates. Because apollo caches items in a flat structure, and doesn't change any fields that aren't returned by the mutation, we don't need to respond with any unchanged fields, and we only need to return the typename + id (or any other information apollo needs for cache identity if you've customized the identity/key for an entity) of any nested entities if they are already in the cache. 
```

Comment:
I felt like this section was a little confusing, and so decided to rewrite it.

I think the crux is that a mutation should ideally return all changed data, and ideally no more. The typical approach is to return the changed type, but I've also seen people create completely custom mutation response types for this.

Also I wanted to call out how apollo actually merges mutation responses back into the cache, because I think it's pretty confusing when first starting out. Often you can just return a typename + id for nested children, because apollo's cache is flat.